src/main/java/org/apache/mesos/scheduler/SchedulerDriverFactory.java

```diff
@@ -0,0 +1,131 @@
+      // User has manually provided a Secret. Provide a Credential with Principal + Secret.
+      // (note: we intentionally avoid logging the content of the credential secret, just in case)
+      LOGGER.info("Creating secret authenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}], credentialSecret[{} bytes]",
+          scheduler, frameworkInfo, masterUrl, credentialSecret.length);
+      credential = Credential.newBuilder()
+          .setPrincipal(getPrincipal(frameworkInfo, "secret"))
+          .setSecretBytes(ByteString.copyFrom(credentialSecret))
+          .build();
+    } else if (isSideChannelActive()) {
+      // Sidechannel auth is enabled. Provide a Credential with only the Principal set.
+      LOGGER.info("Creating sidechannel authenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}]",
+          scheduler, frameworkInfo, masterUrl);
+      credential = Credential.newBuilder()
+          .setPrincipal(getPrincipal(frameworkInfo, "sidechannel"))
+          .build();
+    } else {
+      // No auth. Provide no credential.
+      LOGGER.info("Creating unauthenticated MesosSchedulerDriver for "
+          + "scheduler[{}], frameworkInfo[{}], masterUrl[{}]",
+          scheduler, frameworkInfo, masterUrl);
+      credential = null;
+    }
+    return createInternal(scheduler, frameworkInfo, masterUrl, credential);
+  }
+
+  /**
+   * Broken out into a separate function to allow testing with custom SchedulerDrivers.
+   */
+  protected SchedulerDriver createInternal(
+      final Scheduler scheduler,
+      final FrameworkInfo frameworkInfo,
+      final String masterUrl,
+      final Credential credential) {
+    if (credential == null) {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterUrl);
+    } else {
+      return new MesosSchedulerDriver(scheduler, frameworkInfo, masterUrl, credential);
+    }
+  }
+
+  /**
+   * Returns whether it appears that side channel auth should be used when creating the
+   * SchedulerDriver. Side channel auth may take the form of Bouncer-based framework authentication
+   * or potentially other methods in the future (eg Kerberos). Broken out into a separate function
+   * to allow easy customization in tests.
+   */
+  protected boolean isSideChannelActive() {
+    return System.getenv(SIDECHANNEL_AUTH_ENV_NAME) != null;
+  }
+
+  /**
+   * Extracts the Principal name from the provided {@link FrameworkInfo}, or throws an
+   * {@link IllegalArgumentException} (mentioning the provided {@code authType}) if the Principal
+   * is unavailable.
+   */
+  private static String getPrincipal(final FrameworkInfo frameworkInfo, final String authType) {
+    if (!frameworkInfo.hasPrincipal() || StringUtils.isEmpty(frameworkInfo.getPrincipal())) {
+      throw new IllegalArgumentException(
+          "Unable to create MesosSchedulerDriver for " + authType + " auth, "
```

Comment:
`authType` seems irrelevant to this getter. Seems like better practice would be for the caller to catch the exception/error/empty-string and generate its own error message.
Similarly, this standalone static method shouldn't need to know about the MesosSchedulerDriver, or that its caller is trying to create one.